SPECS/td-agent/td-agent.spec

```diff
@@ -0,0 +1,290 @@
+BuildRequires:  ncurses-devel
+BuildRequires:  openssl-devel
+BuildRequires:  pkg-config
+BuildRequires:  ruby
+BuildRequires:  rubygem-addressable
+BuildRequires:  rubygem-async
+BuildRequires:  rubygem-async-http
+BuildRequires:  rubygem-async-io
+BuildRequires:  rubygem-async-pool
+BuildRequires:  rubygem-aws-eventstream
+BuildRequires:  rubygem-aws-partitions
+BuildRequires:  rubygem-aws-sdk-core
+BuildRequires:  rubygem-aws-sdk-kms
+BuildRequires:  rubygem-aws-sdk-s3
+BuildRequires:  rubygem-aws-sdk-sqs
+BuildRequires:  rubygem-aws-sigv4
+BuildRequires:  rubygem-bundler
+BuildRequires:  rubygem-concurrent-ruby
+BuildRequires:  rubygem-console
+BuildRequires:  rubygem-cool.io
+BuildRequires:  rubygem-digest-crc
+BuildRequires:  rubygem-elasticsearch
+BuildRequires:  rubygem-elasticsearch-api
+BuildRequires:  rubygem-elasticsearch-transport
+BuildRequires:  rubygem-excon
+BuildRequires:  rubygem-faraday
+BuildRequires:  rubygem-ffi
+BuildRequires:  rubygem-fiber-local
+BuildRequires:  rubygem-fluent-config-regexp-type
+BuildRequires:  rubygem-fluent-logger
+BuildRequires:  rubygem-fluent-plugin-elasticsearch
+BuildRequires:  rubygem-fluent-plugin-kafka
+BuildRequires:  rubygem-fluent-plugin-prometheus
+BuildRequires:  rubygem-fluent-plugin-prometheus_pushgateway
+BuildRequires:  rubygem-fluent-plugin-record-modifier
+BuildRequires:  rubygem-fluent-plugin-rewrite-tag-filter
+BuildRequires:  rubygem-fluent-plugin-s3
+BuildRequires:  rubygem-fluent-plugin-systemd
+BuildRequires:  rubygem-fluent-plugin-td
+BuildRequires:  rubygem-fluent-plugin-webhdfs
+BuildRequires:  rubygem-fluentd
+BuildRequires:  rubygem-hirb
+BuildRequires:  rubygem-http_parser.rb
+BuildRequires:  rubygem-httpclient
+BuildRequires:  rubygem-jmespath
+BuildRequires:  rubygem-ltsv
+BuildRequires:  rubygem-mini_portile2
+BuildRequires:  rubygem-msgpack
+BuildRequires:  rubygem-multi_json
+BuildRequires:  rubygem-multipart-post
+BuildRequires:  rubygem-nio4r
+BuildRequires:  rubygem-nokogiri
+BuildRequires:  rubygem-oj
+BuildRequires:  rubygem-parallel
+BuildRequires:  rubygem-prometheus-client
+BuildRequires:  rubygem-protocol-hpack
+BuildRequires:  rubygem-protocol-http
+BuildRequires:  rubygem-protocol-http1
+BuildRequires:  rubygem-protocol-http2
+BuildRequires:  rubygem-public_suffix
+BuildRequires:  rubygem-quantile
+BuildRequires:  rubygem-rake
+BuildRequires:  rubygem-rdkafka
+BuildRequires:  rubygem-ruby-kafka
+BuildRequires:  rubygem-ruby-progressbar
+BuildRequires:  rubygem-ruby2_keywords
+BuildRequires:  rubygem-rubyzip
+BuildRequires:  rubygem-serverengine
+BuildRequires:  rubygem-sigdump
+BuildRequires:  rubygem-strptime
+BuildRequires:  rubygem-systemd-journal
+BuildRequires:  rubygem-td
+BuildRequires:  rubygem-td-client
+BuildRequires:  rubygem-td-logger
+BuildRequires:  rubygem-timers
+BuildRequires:  rubygem-tzinfo
+BuildRequires:  rubygem-tzinfo-data
+BuildRequires:  rubygem-webhdfs
+BuildRequires:  rubygem-yajl-ruby
+BuildRequires:  rubygem-zip-zip
```

Comment:
Are all of these BuildRequires and what looks like the same set of Requires required?

Reply:
I have removed indirect dependency rubygems from both BuildRequires and Requires. The remaining ones are defined in td-agent's source code, so they will definitely be used when building the packages as well as during the service runtime.